app/Console/Commands/GenerateResource.php

```diff
@@ -29,14 +29,44 @@ class GenerateResource extends Command
     public function handle()
     {
         list($resource, $table) = [$r = ucfirst($this->argument('resource')), snake_case(str_plural($r))];
-        $this->call('make:model', ['name' => $resource]);
-        $this->call('make:controller', ['name' => 'Admin/'. $resource . 'Controller']);
-        $this->call('make:controller', ['name' => 'Api/DataTables/'. $resource . 'DataTable']);
-        $this->call('make:migration', ['name' => 'create_' . $table . '_table']);
-        $this->call('make:request', ['name' => 'Admin/' . $resource . 'Request']);
-        $this->call('make:form', ['name' => 'Forms/Admin/' . $resource . 'Form']);
-        $this->createDirectory($table);
-        $this->info('New resource generated successfully');
+
+        $options = collect($this->options())->only([
```

Comment:
Why don't you store actions in array too, so you may get rid from all the if clauses below.

```php
 $options = [
    ['model'      => ['name' => $resource]],
    ['controller' => ['name' => 'Admin/'. $resource . 'Controller']]
    ....
];
```

First you can check if generateAllStuff is true, if yes then you loop through all the options and call, if not you just loop through the valid ones, given by user input?

Reply:
Yes, it will be cleaner.
I'll do this.